src/reconstruction/enzyme_constrained.jl

```diff
@@ -29,20 +36,34 @@ ecmodel = make_enzyme_constrained_model(
         "total" => 0.5,
     ),
 )
-```
 
+ecmodel2 = make_enzyme_constrained_model(
+    model;
+    total_enzyme_capacity = 0.5
+) 
+```
 # Notes
 Reactions with no turnover number data, or non-enzymatic reactions that should
-be ignored, must have `nothing` in the `gene_associations` field of the
+be ignored, *must* have `nothing` in the `gene_associations` field of the
 associated reaction.
 """
 function make_enzyme_constrained_model(
     model::AbstractMetabolicModel;
-    gene_product_mass_group::Dict{String,Vector{String}} = Dict(
-        "uncategorized" => genes(model),
-    ),
-    gene_product_mass_group_bound::Dict{String,Float64} = Dict("uncategorized" => 0.5),
+    gene_product_mass_group::Maybe{Dict{String,Vector{String}}} = nothing,
+    gene_product_mass_group_bound::Maybe{Dict{String,Float64}} = nothing,
+    total_enzyme_capacity::Maybe{Float64} = nothing,
 )
+    if isnothing(gene_product_mass_group) &&
+       isnothing(gene_product_mass_group_bound) &&
+       !isnothing(total_enzyme_capacity)
+        gene_product_mass_group = Dict("uncategorized" => genes(model))
+        gene_product_mass_group_bound = Dict("uncategorized" => total_enzyme_capacity)
+    elseif isnothing(gene_product_mass_group) &&
+           isnothing(gene_product_mass_group_bound) &&
+           isnothing(total_enzyme_capacity)
+        throw(ArgumentError("Invalid combination of keyword arguments."))
+    end
```

Comment:
This is not very readable and actually misses one problematic case (gpmg is set but gpmgb is not). The usual way is to patch the arguments towards a good shape and only have a simple check on whether the good shape was reached. This way you'd literally have to check 3^2 combinations of `isnothing` vs `!isnothing`.

```suggestion
    if !isnothing(total_enzyme_capacity)
        gene_product_mass_group = Dict("uncategorized" => genes(model))
        gene_product_mass_group_bound = Dict("uncategorized" => total_enzyme_capacity)
    end
    isnothing(gene_product_mass_group) && throw(ArgumentError("missing mass group specification"))
    isnothing(gene_product_mass_group_bound) && throw(ArgumentError("missing mass group bounds"))
```